exonum/src/storage/list_index.rs

```diff
@@ -581,6 +639,23 @@ mod tests {
             let mut list_index = ListIndex::new_in_family(IDX_NAME, &vec![01], &mut fork);
             super::list_index_iter(&mut list_index);
         }
+
+        #[test]
+        fn list_index_clear_in_family() {
+            const PARAMS: &[(u32, u32, bool)] = &[
+                (0_u32, 5_u32, false),
+                (5_u32, 0_u32, false),
+                (1_u32, 7_u32, true),
+                (7_u32, 1_u32, true),
+            ] as &[(u32, u32, bool)];
+
+            for &(x, y, merge_before_clear) in PARAMS {
+                let dir = TempDir::new(super::gen_tempdir_name().as_str()).unwrap();
```

Comment:
I guess we don't need to call `gen_tempdir_name()` here. We can simply use a string `"list_index_clear_in_family"` or something. `TempDir::new()` uses the prefix argument to generate a unique random name for the temporary directory so generating a random prefix for it is a bit excessive.

Ditto below in `rocksdb_tests`.

Reply:
Is this specific to the tests I've added, or are you proposing to refactor all tests in `storage::list_index` module?

Reply:
@ilammy there was some problems with generated unique random name by `TempDir::new()`.

Reply:
@aleksuss, hm... I'm curious, any references to that? TempDir [seems to do exactly the same thing](https://docs.rs/tempdir/0.3.7/src/tempdir/lib.rs.html#200-217) with generating random names, giving up only after 2^31 retries. So I find it really strange that it fails to work.